shared/python/src/xayn_infra_lib/generation.py

```
@@ -0,0 +1,110 @@
+from abc import ABC, abstractmethod
+from typing import List
+
+import json
+
+
+class LLMProvider(ABC):
+    @abstractmethod
+    def generate(self, prompt: str) -> str:
+        pass
```

Not something quite important, but we could raise an exception here if somehow this function is called:
```
raise NotImplementedError("Subclasses must override the generate() method.")
```